### PR TITLE
support explainability for public node square

### PR DESCRIPTION
**Pull request by haim-kermany**
the solution is to always create a "public network" icon.
it holds the "public network" explainability information
if there is no connection, do not show in on the canvas.

### CODE REVIEW

File: pkg/vpcmodel/drawioOutput.go [hunk 1]
```diff
@@ -113,6 +113,29 @@ func (d *DrawioOutputFormatter) createNodes() {
 			}
 		}
 	}
+	d.createPublicNetworkIcon()
+}
+
+// createPublicNetworkIcon() - will always create the the iconTreeNode that represent the Public Network
+// in case that it has no connection, we will not show it in drawio
+// however, we always create it since it's needed for holding the explainability information of the public network square
```
Comment (ShiriMoran) [inline pkg/vpcmodel/drawioOutput.go:121]:
```suggestion
// createPublicNetworkIcon() - creates the iconTreeNode that represents the Public Network (always)
// it holds the explainability information of the public network square, which may be connected or not connected
// to the internal endpoints
```

File: pkg/vpcmodel/multiExplainability.go [hunk 2]
```diff
@@ -169,6 +169,9 @@ func collectNodesForExplanation(cConfigs *MultipleVPCConfigs, conns map[string]*
 			}
 		}
 	}
+	if cConfigs.publicNetworkNode != nil {
```
Comment (ShiriMoran) [inline pkg/vpcmodel/multiExplainability.go:172]: What is the situation in which it is legit  cConfigs.publicNetworkNode will be nil?
Shouldn't this node be always present?

File: pkg/vpcmodel/multi_vpc_config.go [hunk 3]
```diff
@@ -20,10 +20,12 @@ type MultipleVPCConfigs struct {
 	configs          map[string]*VPCConfig // a map from the vpc resource uid to the vpc config
 	toCompareConfigs map[string]*VPCConfig // a map from the vpc resource uid to the vpc config that we want to compare
 	provider         collector_common.Provider
+	// publicNetworkNode - the EndpointElem that represent all the cidr which are not in ant vpc:
```
Comment (ShiriMoran) [inline pkg/vpcmodel/multi_vpc_config.go:23]:
```suggestion
	// publicNetworkNode - the EndpointElem that  entire external cidr which is in any vpc:
```

**Review by ShiriMoran — COMMENTED** [review]
LGTM
I did not review the `drawio` related files

File: pkg/drawio/connectivityMap.svg.tmpl
Comment (zivnevo) [inline pkg/drawio/connectivityMap.svg.tmpl:1]: In line 233, please change the text to:
`Explaining connectivity from <src> to <dst>`

**Review by zivnevo — CHANGES_REQUESTED** [review]
A few small things

File: pkg/drawio/connectivityMap.svg.tmpl
Comment (haim-kermany) [reply pkg/drawio/connectivityMap.svg.tmpl:1]: fixed

File: pkg/vpcmodel/drawioOutput.go [hunk 4]
```diff
@@ -113,6 +113,29 @@ func (d *DrawioOutputFormatter) createNodes() {
...
+		d.gen.publicNetwork.PublicNetworkIcon = d.gen.TreeNode(d.cConfigs.publicNetworkNode)
+		// check if to show the icon, only if it is a src/dst:
+		publicIconHasConnection := false
+		for _, vpcConn := range d.gConns {
+			if slices.IndexFunc(vpcConn.GroupedLines, func(line *groupedConnLine) bool {
+				return line.Src == d.cConfigs.publicNetworkNode || line.Dst == d.cConfigs.publicNetworkNode
+			}) >= 0 {
+				publicIconHasConnection = true
+				break
+			}
```
[the remark this answers is not in the document]
Comment (haim-kermany) [reply pkg/vpcmodel/drawioOutput.go]: fixed

File: pkg/vpcmodel/drawioOutput.go [hunk 5]
```diff
@@ -113,6 +113,29 @@ func (d *DrawioOutputFormatter) createNodes() {
...
 		}
 	}
+	d.createPublicNetworkIcon()
+}
+
+// createPublicNetworkIcon() - will always create the the iconTreeNode that represent the Public Network
+// in case that it has no connection, we will not show it in drawio
+// however, we always create it since it's needed for holding the explainability information of the public network square
+func (d *DrawioOutputFormatter) createPublicNetworkIcon() {
+	if d.cConfigs.publicNetworkNode != nil {
```
[the remark this answers is not in the document]
Comment (haim-kermany) [reply pkg/vpcmodel/drawioOutput.go]: fixed

**Review by zivnevo — APPROVED** [review]

**Pull request merged**